src/aliases/store/models/alias.go

```diff
@@ -15,29 +22,64 @@ type Alias struct {
 	Value AliasValue
 }
 
-func AliasFromEntity(ent aliasent.Alias) Alias {
+func checkAliasValue(value aliasent.AliasValue) error {
+	for _, v := range value {
+		// we just want the decoding to happen without error
+		_, err := base64.StdEncoding.DecodeString(v)
+		if err != nil {
+			return errors.InvalidFormatError("alias value is incorrect: %v: %v", err, v)
+		}
+	}
+	return nil
+}
+
+func AliasFromEntity(ent aliasent.Alias) (alias Alias, err error) {
+	// check that the ent.Value only contains base64 string
+	err = checkAliasValue(ent.Value)
+	if err != nil {
+		return alias, err
+	}
+
+	var b bytes.Buffer
+	err = json.NewEncoder(&b).Encode(ent.Value)
+	if err != nil {
+		return alias, err
+	}
 	return Alias{
 		Key:          AliasKey(ent.Key),
 		RegistryName: RegistryName(ent.RegistryName),
-		Value:        AliasValue(ent.Value),
-	}
+		Value:        AliasValue(b.String()),
+	}, nil
 }
 
-func (a *Alias) ToEntity() *aliasent.Alias {
+func (a *Alias) ToEntity() (*aliasent.Alias, error) {
+	var value aliasent.AliasValue
+	err := json.Unmarshal([]byte(a.Value), &value)
+	if err != nil {
+		return nil, err
+	}
+
+	err = checkAliasValue(value)
+	if err != nil {
+		return nil, err
+	}
 	return &aliasent.Alias{
 		Key:          aliasent.AliasKey(a.Key),
 		RegistryName: aliasent.RegistryName(a.RegistryName),
-		Value:        aliasent.AliasValue(a.Value),
-	}
+		Value:        value,
+	}, nil
 }
 
-func AliasesToEntity(aliases []Alias) []aliasent.Alias {
+func AliasesToEntity(aliases []Alias) ([]aliasent.Alias, error) {
```

Comment:
This function does not belong here. It seems to be a utility function of a specific use case.

Reply:
It's a utility func that has its place here as it only manipulates the models types from/to entities.